Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -116,6 +118,17 @@ private void IgniteOnCollide(EntityUid uid, IgniteOnCollideComponent component,
                 RemCompDeferred<IgniteOnCollideComponent>(uid);
         }
 
+        private void IgniteOnHit(EntityUid uid, IgniteOnHitComponent component, ref ProjectileHitEvent args)
+        {
+            var otherEnt = args.Target;
+
+            if (!EntityManager.TryGetComponent(otherEnt, out FlammableComponent? flammable))
+                return;
+
+            flammable.FireStacks += component.FireStacks;
+            Ignite(otherEnt, uid, flammable);
+        }
+
```

Comment:
This should just be using igniteoncollide as there's no good reason to have another component.

Reply:
removed the component and added a check for the fly-by fixture